nkms/blockchain/eth/interfaces.py

```diff
@@ -141,12 +141,12 @@ def dump_contract(self, address: str=None) -> dict:
         If no contract is found, it will raise NoKnownContract.
         """
         chain_data = self.dump_chain()
-        try:
-            contract_data = chain_data[address]
-        except KeyError:
-            raise self.UnknownContract('No known contract with address {}'.format(address))
-        else:
-            return contract_data
+        if address in chain_data:
```

Comment:
@tux and I deliberated about how to store this data - Does it make sense to have addresses as dict keys? What happens in the case of multiple deployments?

Reply:
For me it is not obvious how to store contracts. This implementation is close to populus. Also populus stores bytecode, but I'm not sure that will help much. When we will deploying multiple same contracts (Dispatcher, for example) then we can't find out wich contract should be returned using only contract name